samcli/commands/validate/validate.py

```diff
@@ -95,3 +114,40 @@ def _read_sam_file(template):
         sam_template = yaml_parse(sam_template.read())
 
     return sam_template
+
+def _lint(ctx, template):
+    """
+    Parses provided SAM template and maps errors from CloudFormation template back to SAM template.
+
+    Parameters
+    -----------
+    ctx
+        Click context object
+    template
+        Path to the template file
+
+    """
+
+    try:
+        if ctx.debug:
+            (args, filenames, formatter) = cfnlint.core.get_args_filenames([template, "--debug"])
```

Comment:
Can we add details why we need special handling of `--debug` flag here?

Reply:
We need a way to pass the `--debug` flag to cfn-lint since cfn-lint also takes in a `--debug- optional parameter as well. Do you mean you would want some comments in the code explaining why?